lib/controllers/pr-reviews-controller.js

```diff
@@ -0,0 +1,156 @@
+import React, {Fragment} from 'react';
+import PropTypes from 'prop-types';
+import {RelayConnectionPropType} from '../prop-types';
+
+import PullRequestReviewCommentsContainer from '../containers/pr-review-comments-container';
+import PullRequestReviewCommentsView from '../views/pr-review-comments-view';
+import {PAGE_SIZE, PAGINATION_WAIT_TIME_MS} from '../helpers';
+
+export default class PullRequestReviewsController extends React.Component {
+  static propTypes = {
+    relay: PropTypes.shape({
+      hasMore: PropTypes.func.isRequired,
+      loadMore: PropTypes.func.isRequired,
+      isLoading: PropTypes.func.isRequired,
+    }).isRequired,
+    pullRequest: PropTypes.shape({
+      reviews: RelayConnectionPropType(
+        PropTypes.object,
+      ),
+    }),
+    getBufferRowForDiffPosition: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.reviewsById = new Map();
+  }
+
+  componentDidMount() {
+    this._attemptToLoadMoreReviews();
+  }
+
+  _attemptToLoadMoreReviews = () => {
+    if (!this.props.relay.hasMore()) {
+      return;
+    }
+
+    if (this.props.relay.isLoading()) {
+      setTimeout(this._loadMoreReviews, PAGINATION_WAIT_TIME_MS);
+    } else {
+      this._loadMoreReviews();
+    }
+  }
+
+  _loadMoreReviews = () => {
+    this.props.relay.loadMore(
+      PAGE_SIZE,
+      error => {
+        /* istanbul ignore if */
+        if (error) {
+          // eslint-disable-next-line no-console
+          console.error(error);
+        } else {
+          this._attemptToLoadMoreReviews();
+        }
+      },
+    );
+  }
+
+  render() {
+    if (!this.props.pullRequest || !this.props.pullRequest.reviews) {
+      return null;
+    }
+
+    const commentThreads = Object.keys(this.state).reverse().map(rootCommentId => {
```

Comment:
Looks like we're reversing the comment thread order time we render.  Seems wasteful. Is there a way to avoid that?